src/main/java/nextstep/security/httpsecurity/HttpSecurity.java

```diff
@@ -0,0 +1,110 @@
+package nextstep.security.httpsecurity;
+
+import jakarta.servlet.Filter;
+import nextstep.security.authentication.AuthenticationManager;
+import nextstep.security.authentication.AuthenticationProvider;
+import nextstep.security.config.Customizer;
+import nextstep.security.config.DefaultSecurityFilterChain;
+import nextstep.security.httpsecurity.configurer.*;
+
+import java.util.*;
+
+public class HttpSecurity {
+    private final LinkedHashMap<Class<? extends SecurityConfigurer>, SecurityConfigurer> configurers = new LinkedHashMap<>();
+    private List<Filter> filters = new ArrayList<>();
+    private final Map<Class<?>, Object> sharedObjects = new HashMap<>();
+
+    public HttpSecurity(AuthenticationManagerBuilder authenticationManagerBuilder, Map<Class<?>, Object> sharedObjects) {
+        setSharedObject(AuthenticationManagerBuilder.class, authenticationManagerBuilder);
+        for (Map.Entry<Class<?>, Object> entry : sharedObjects.entrySet()) {
+            setSharedObject((Class<Object>) entry.getKey(), entry.getValue());
+        }
+    }
+
+    public HttpSecurity securityContext(Customizer<SecurityContextConfigurer> securityContextCustomizer) {
+        securityContextCustomizer.customize(getOrApply(new SecurityContextConfigurer()));
+        return HttpSecurity.this;
```

Comment:
```suggestion
        return this;
```
별개의 이야기이지만 `this` 하나만으로도 본인을 반환한다는 의미를 충실히 전달할 수 있기 때문에 제거해도 된다고 생각합니다 :)